backend/src/main/controller/EnumController.ts

```diff
@@ -0,0 +1,64 @@
+import { DiscountType } from '../data/DiscountType';
+import { PromotionCategory } from '../data/PromotionCategory';
+import { CuisineType } from '../data/CuisineType';
+import { NextFunction, Request, Response } from 'express';
+import Joi from 'joi';
+
+enum SupportedEnums {
+  DiscountType = 'DiscountType',
+  PromotionCategory = 'PromotionCategory',
+  CuisineType = 'CuisineType',
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+export class EnumController {
+  // validates that incoming string is of type SupportedEnums
+  private enumSchemaValidation = Joi.string()
+    .valid(...Object.values(SupportedEnums))
+    .required();
+
+  /**
+   * returns all possible values of an enum. Otherwise throws validation error if not a supported enum
+   * */
+  getEnum = async (
+    request: Request,
+    response: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    try {
+      const enumString: SupportedEnums = await this.enumSchemaValidation.validateAsync(
+        request.params.enum,
+        { abortEarly: false }
+      );
+      const enumClass = this.getEnumClass(enumString);
+      const enumValues = this.getEnumValues(enumClass);
+      return response.status(200).send(enumValues);
+    } catch (e) {
+      next(e);
+    }
+  };
```

Comment:
@andrewzulaybar I remember you said you wanted enums to ideally be exposed through endpoints.

This controller now allows us to call `enums/:enum` and will give us all the possible values of the specified enum.

So we have
```
GET enum/DiscountType
GET enum/CuisineType
GET enum/PromotionCategory
```

Reply:
This is great, thank you 🙌